### PR TITLE
feat: extend developer workflow docs for new wash commands

### PR DESCRIPTION
**Pull request by vados-cosmonic**
## Feature or Problem

This commit extends the developer workflow docs with the new commands that were added to wash and released in version 0.29.0:

- `wash app validate`
- `wash up --manifest`
- `wash app deploy --replace`

These commands make the workflow quicker (and are the precursors to `wash app dev`), so having some docs around them will hopefully improve DX.

### CODE REVIEW

File: docs/cli/app.md [hunk 1]
```diff
@@ -77,6 +77,12 @@ wash app deploy <name> [version]
 wash app deploy petclinic v0.0.1
 ```
 
+You can also *replace* a WADM application manifest with the `--replace` option:
```
Comment (ericgregory) [inline docs/cli/app.md]:
```suggestion
You can also *replace* a wadm application manifest with the `--replace` option:
```

File: docs/cli/app.md [hunk 2]
```diff
@@ -87,44 +93,52 @@ wash app undeploy <name>
 wash app undeploy petclinic
 ```
 
+### `validate`
+Figure out if a WADM manifest has any issues that might prevent it from working properly.
```
Comment (ericgregory) [inline docs/cli/app.md]:
```suggestion
Figure out if a wadm manifest has any issues that might prevent it from working properly.
```

File: docs/cli/up.md [hunk 3]
```diff
@@ -180,3 +180,7 @@ If enabled, wasmCloud will not be downloaded if it's not installed
 wadm version to download, e.g. `v0.4.0`. See https://github.com/wasmCloud/wadm/releases for releases [env: WADM_VERSION=] [default: v0.6.0]
 
 #### --disable-wadm
+
+#### --wadm-manifest
+
+Specify a WADM application manifest to run right after starting the host
```
Comment (ericgregory) [inline docs/cli/up.md]:
```suggestion
Specify a wadm application manifest to run right after starting the host.
```

File: docs/developer/workflow.md [hunk 4]
```diff
@@ -64,16 +64,62 @@ Previous guides used `wash reg push`, which is now deprecated and will be remove
...
 
-### Purging the OCI cache
+## Purging the OCI cache
 
 The wasmCloud host runtime caches the files that it receives from OCI registries beneath whatever `temp` directory your operating system prefers. Because images in an OCI registry are supposed to be _immutable_ (another reason we recommend against using `latest` when requesting an image version), the wasmCloud host has no reason to automatically purge or overwrite these files in the cache.
 
 During your local development iterations, you will likely find yourself pushing the same file with the same OCI reference over and over again. In order for the wasmCloud host to see these changes, you'll need to _drain_ the wasmCloud host cache. This can be done by executing one of the variants of `wash drain`, such as `wash drain all`.
+
+## Testing WADM manifests faster with `wash`
```
Comment (ericgregory) [inline docs/developer/workflow.md]:
```suggestion
## Testing wadm manifests faster with `wash`
```

File: docs/developer/workflow.md [hunk 5]
```diff
@@ -64,16 +64,62 @@ Previous guides used `wash reg push`, which is now deprecated and will be remove
...
+
+As of `wash` v0.29.0, deploying applications and iterative development has become much easier with the help of a few new commands.
+
+### Validate WADM application manifests before deploying them
```
Comment (ericgregory) [inline docs/developer/workflow.md]:
```suggestion
### Validate wadm application manifests before deploying them
```

File: docs/developer/workflow.md [hunk 6]
```diff
@@ -64,16 +64,62 @@ Previous guides used `wash reg push`, which is now deprecated and will be remove
...
+
+Misspelled interfaces, missing links, and other issues can be hard to find in a WADM application manifest.
```
Comment (ericgregory) [inline docs/developer/workflow.md]:
```suggestion
Misspelled interfaces, missing links, and other issues can be hard to find in a wadm application manifest.
```

File: docs/developer/workflow.md [hunk 7]
```diff
@@ -64,16 +64,62 @@ Previous guides used `wash reg push`, which is now deprecated and will be remove
...
+
+To find potential issues *before* you `wash app deploy` your manifest, use `wash app validate`:
+
+```console
+wash app validate path/to/your/wadm.yaml
+```
+
+`wash app validate` will check for common issues and errors in your WADM manifest, and notify you of any issues.
```
Comment (ericgregory) [inline docs/developer/workflow.md]:
```suggestion
`wash app validate` will check for common issues and errors in your wadm manifest, and notify you of any issues.
```

File: docs/developer/workflow.md [hunk 8]
```diff
@@ -64,16 +64,62 @@ Previous guides used `wash reg push`, which is now deprecated and will be remove
...
+
+### Start a host and deploy a WADM application manifest in one command
```
Comment (ericgregory) [inline docs/developer/workflow.md]:
```suggestion
### Start a host and deploy a wadm application manifest in one command
```

File: docs/developer/workflow.md [hunk 9]
```diff
@@ -64,16 +64,62 @@ Previous guides used `wash reg push`, which is now deprecated and will be remove
...
+
+The easiest way to test a WADM application is to start a host & deploy the application, and start using it.
```
Comment (ericgregory) [inline docs/developer/workflow.md]:
```suggestion
The easiest way to test a wadm application is to start a host, deploy the application, and start using it.
```

File: docs/developer/workflow.md [hunk 10]
```diff
@@ -64,16 +64,62 @@ Previous guides used `wash reg push`, which is now deprecated and will be remove
...
+
+To quickly run a host and deploy a *single* WADM manifest, run `wash up` with `--wadm-manifest`:
```
Comment (ericgregory) [inline docs/developer/workflow.md]:
```suggestion
To quickly run a host and deploy a *single* wadm manifest, run `wash up` with `--wadm-manifest`:
```

File: docs/developer/workflow.md [hunk 11]
```diff
@@ -64,16 +64,62 @@ Previous guides used `wash reg push`, which is now deprecated and will be remove
...
+
+```console
+wash up --wadm-manifest path/to/your/wadm.yaml
+```
+
+When run this way, `wash up` will start a single host lattice as normal, but *also* deploy your provided manifest to the running lattice, so you can interact with your application as soon as possible.
+
+### Easily deploy a WADM application manifest continuously
```
Comment (ericgregory) [inline docs/developer/workflow.md]:
```suggestion
### Easily deploy a wadm application manifest continuously
```

**Review by ericgregory — APPROVED** [review]

**Comment by ericgregory** [comment]
Just the nittiest of nits for stylistic consistency—looks great 🚀

**Comment by vados-cosmonic** [comment]
Thanks @ericgregory  for making it so easy to apply the changes!

**Pull request merged**